src/pg_to_evalscript/javascript_processes/all.js

```diff
@@ -0,0 +1,38 @@
+function all(arguments) {
+  const { data, ignore_nodata = true } = arguments;
+
+  if (data === undefined) {
+    throw new Error("Mandatory argument `data` is not defined.");
+  }
+
+  if (!Array.isArray(data)) {
+    throw new Error("Argument `data` is not an array.");
+  }
+
+  if (!data.every((x) => typeof x === "boolean" || x === null)) {
+    throw new Error(
+      "Values in argument `data` can only be of type boolean or null."
+    );
+  }
+
+  if (ignore_nodata) {
+    if (data.filter((x) => x !== null).length === 0) {
+      return null;
+    }
+    return data.filter((x) => x !== null).every((x) => x === true);
+  }
+
+  if (data.length === 0) {
+    return null;
+  }
```

Comment:
This `if` and 

```
if (data.filter((x) => x !== null).length === 0) {
   null;
}
```
 in the `if (ignore_nodata) {` can be merged and put before the `if (ignore_nodata) {` since it should be the same behaviour regardless if `ignore_nodata` is `true` or `false`.

Reply:
Should be fixed